### PR TITLE
[KYUUBI #2247] Change log4j2 properties to xml

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

- change log4j2-test.properties to log4j2-test.xml
- add the unit test log4j2.xml for spark relative submodule, and remove the log4j.properties

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-extension-spark-3-1/src/test/resources/log4j2-test.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<!-- Extra logging related to initialization of Log4j. 
+ Set to debug or trace if log4j initialization is failing. -->
+<Configuration status="warn">
+  <Appenders>
+   <Console name="STDOUT" target="SYSTEM_OUT">
```
Comment (pan3793) [inline extensions/spark/kyuubi-extension-spark-3-1/src/test/resources/log4j2-test.xml:23]: Have you reformat the xml? The indent looks wierd

Comment (cfmcgrady) [reply extensions/spark/kyuubi-extension-spark-3-1/src/test/resources/log4j2-test.xml:23]: Seems we don't have a dev script to help us format XML files, which style do you like? 4 spaces or 2 spaces?

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-3-1/src/test/resources/log4j2-test.xml:23]: Our `pom.xml` uses 4 spaces

**Comment by cfmcgrady** [comment]
cc @jiaoqingbo @deadwind4

**Review by jiaoqingbo — APPROVED** [review]

**Review by deadwind4 — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

File: pom.xml [hunk 2]
```diff
@@ -1601,8 +1601,7 @@
                         </environmentVariables>
                         <systemProperties>
                             <log4j.ignoreTCL>true</log4j.ignoreTCL>
-                            <log4j.configuration>file:src/test/resources/log4j.properties</log4j.configuration>
-                            <log4j2.configurationFile>file:src/test/resources/log4j2-test.properties</log4j2.configurationFile>
+                            <log4j2.configurationFile>file:src/test/resources/log4j2-test.xml</log4j2.configurationFile>
```
Comment (yaooqinn) [inline pom.xml:1604]: it seems shall be `<log4j2.configurationFile>src/test/resources/log4j2-test.xml</log4j2.configurationFile>`


```scala
WARN Invalid URI file:src/test/resources/log4j2-test.xml
```

Comment (yaooqinn) [reply pom.xml:1604]: cc @cfmcgrady @pan3793

Comment (pan3793) [reply pom.xml:1604]: cc @1456173400  would you please take a look?

Comment (turboFei) [reply pom.xml:1604]: fixed in #2971

**Pull request closed without merging**